src/sentry/api/endpoints/organization_events_trends_v2.py

```diff
@@ -218,22 +227,25 @@ def get_event_stats_metrics(_, user_query, params, rollup, zerofill_results, __)
                 int(request.GET.get("topEvents", DEFAULT_TOP_EVENTS_LIMIT)),
                 MAX_TOP_EVENTS_LIMIT,
             )
+
             # Fetch transactions names with the highest event count
-            top_events = get_top_events(
+            global top_trending_transactions
```

Comment:
If you need global state, you should use redis. When this gets deployed, this global variable will get overwritten by many requests. It will also be different depending on which API pod you hit

Reply:
My question would be though, do you need global state here?

Reply:
from what I can tell, the use case here is so that `top_trending_transactions` is saved and usable from another part of the processing later.

So it'd be better to declare `top_trending_transactions` like it is right now in the outer scope and use `nonlocal` here. Using `global` here will not use the variable declared in the outer scope but rather create a new variable in the global scope that is shared which can then lead to race conditions.

Reply:
yeah sorry I wanted to use `nonlocal` instead